File: src/com/cqut/service/node/NodeService.java
package com.cqut.service.node;

import java.util.List;

import com.cqut.entity.Node;

public interface NodeService {
	
	public int insertNode(Node node);

	public int deleteNode(int nodeId);
	
	public int updateNode(Node node);
	
	public List<Node> getNodes();
	
	public Node getNodeById(int EId);
	
	public List<Node> getNodesByModuleId(int moduleId);
	
	public List<Node> getNodesByModuleIdAndName(int moduleId,String name);
	
	public List<Node> getNodesName(String name);
}

File: WebContent/cqut/hrms/view/node/js/NodeService.js
var nodeApp = angular.module("nodeApp",[]);

nodeApp.factory('NodeService',['$resource',function($resource){
	
	var nodeService = $resource('/Node/:action',{});
	
	nodeService.insertNode= function(node,sucesscb,errorcb){
		
		nodeService.save({action:'insertNode'},node,sucesscb,errorcb);
	}
	
	nodeService.updateNode = function(node,sucesscb,errorcb){
		
		nodeService.save({action:'updateNode'},node,sucesscb,errorcb);
	}
	
	nodeService.deleteNode = function(nodeId,sucesscb,errorcb){
		
		nodeService.save({action:'deleteNode'},nodeId,sucesscb,errorcb);
	}
	
	nodeService.getNodes = function(sucesscb,errorcb){
		
		nodeService.query({action:'getNodes'},sucesscb,errorcb);	
	}
	
	nodeService.getNodeById=function(nodeId,sucesscb,errorcb){
		nodeService.save({action:'getNodeById'},nodeId,sucesscb,errorcb)
	}
	
	nodeService.getNodesByModuleId=function(moduleId,sucesscb,errorcb){
		
		nodeService.save({action:'getNodesByModuleId'},{moduleId:moduleId},sucesscb,errorcb)
	}

	nodeService.getNodesByModuleIdAndName=function(id,name,sucesscb,errorcb){
		
		nodeService.save({action:'getNodesByModuleIdAndName'},{EId:id,name:name},sucesscb,errorcb);
	}

	nodeService.getNodesByName=function(name,sucesscb,errorcb){
		
		nodeService.save({action:'getNodesByName'},{name:name},sucesscb,errorcb);
	}
	return nodeService;
}]);